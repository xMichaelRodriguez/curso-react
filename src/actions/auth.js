import Swal from "sweetalert2";
import { fetchAsync, fetchSync } from "../helpers/fetch";
import { types } from "../types/types";
import { eventLogout } from "./events";

export const startLogin = (email, password) => {
  return async (dispatch) => {
    Swal.fire({
      title: "Iniciando Sesion...",
      text: "Por favor espere...",
      allowOutsideClick: false,
      allowEnterKey: false,
      allowEscapeKey: false,
      showConfirmButton: false,
      onBeforeOpen: () => {
        Swal.showLoading();
      },
    });
    const resp = await fetchSync("auth", { email, password }, "POST");

    const body = await resp.json();

    if (body.ok) {
      localStorage.setItem("token", body.token);
      localStorage.setItem("token-initDate", new Date().getTime());
      dispatch(login({ uid: body.uid, name: body.name }));
      Swal.close();
    } else {
      Swal.close();
      Swal.fire("Error", body.msg, "error");
    }
  };
};

export const startRegister = (name, email, password) => {
  return async (dispatch) => {
    Swal.fire({
      title: "Registrando...",
      text: "Por favor espere...",
      allowOutsideClick: false,
      allowEnterKey: false,
      allowEscapeKey: false,
      showConfirmButton: false,
      onBeforeOpen: () => {
        Swal.showLoading();
      },
    });
    const resp = await fetchSync("auth/new", { name, email, password }, "POST");

    const body = await resp.json();

    if (body.ok) {
      localStorage.setItem("token", body.token);
      localStorage.setItem("token-initDate", new Date().getTime());
      dispatch(login({ uid: body.uid, name: body.name }));
      Swal.close();
    } else {
      Swal.close();
      Swal.fire("Error", body.msg, "error");
    }
  };
};

export const startChecking = () => {
  return async (dispatch) => {
    const resp = await fetchAsync("auth/renew");

    const body = await resp.json();

    if (body.ok) {
      localStorage.setItem("token", body.token);

      localStorage.setItem("token-initDate", new Date().getTime());

      dispatch(login({ uid: body.uid, name: body.name }));

      Swal.close();
    } else {
      dispatch(checkingFish());
    }
  };
};

const checkingFish = () => ({
  type: types.authFinishChecking,
});

const login = (user) => ({
  type: types.authLogin,
  payload: user,
});

export const startLogout = () => {
  return (dispatch) => {
    localStorage.clear();

    dispatch(eventLogout());

    dispatch(logout());
  };
};

const logout = () => ({
  type: types.authLogout,
});
