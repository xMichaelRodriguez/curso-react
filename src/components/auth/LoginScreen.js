import React from "react";
import { FaSignInAlt } from "react-icons/fa";
import { Link } from "react-router-dom";
export const LoginScreen = () => {
  const handleLogin = (e) => e.preventDefault();
  return (
    <div className="shadow auth">
      <form onSubmit={handleLogin}>
        <h1 className="mb-4">
          <ins>Blog</ins>
        </h1>

        <div className="form-row">
          <div className="col-md-12 mb-3">
            <div className="form-group">
              <label>Email</label>
              <input
                className="form-control"
                type="email"
                placeholder="example@dominio.com"
              />
            </div>
          </div>
          <div className="col-md-12 mb-3">
            <label>Password</label>
            <input
              type="password"
              className="form-control"
              placeholder="Password"
            />
          </div>
          <div className="col-md-6 mb-3">
            <button className="btn primary btn-lg" type="submit">
              <FaSignInAlt /> &nbsp;
              <span>Login</span>
            </button>
          </div>
          <div className="col-md-6 w-100">
            <Link to="/auth/register" class="btn btn-link text-capitalize">
              forgot password?
            </Link>
          </div>
        </div>
      </form>
    </div>
  );
};
