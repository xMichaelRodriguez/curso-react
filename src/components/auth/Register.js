import React from 'react';
import { FaSignInAlt } from 'react-icons/fa';
import { Link } from 'react-router-dom';

export const Register = () => {
  const handleRegister = (e) => e.preventDefault();
  return (
    <div className='shadow auth animate__animated animate__fadeIn'>
      <form onSubmit={handleRegister}>
        <h1 className='mb-4'>
          <ins>Blog</ins>
        </h1>

        <div className='form-row'>
          <div className='col-md-6'>
            <div className='form-group'>
              <label>First Name</label>
              <input
                className='form-control'
                type='text'
                placeholder='First Name'
              />
            </div>
          </div>
          <div className='col-md-6'>
            <div className='form-group'>
              <label>Last Name</label>
              <input
                className='form-control'
                type='text'
                placeholder='Last Name'
              />
            </div>
          </div>
          <div className='col-md-12 mb-3'>
            <div className='form-group'>
              <label>Email</label>
              <input
                className='form-control'
                type='email'
                placeholder='example@dominio.com'
              />
            </div>
          </div>
          <div className='col-md-12 mb-3'>
            <label>Password</label>
            <input
              type='password'
              className='form-control'
              placeholder='Password'
            />
          </div>
          <div className='col-md-6 mb-3'>
            <button className='btn primary btn-lg' type='submit'>
              <FaSignInAlt /> &nbsp;
              <span>Register</span>
            </button>
          </div>
          <div className='col-md-6 w-100'>
            <Link to='/auth/' class='btn btn-link text-capitalize'>
              I have an account
            </Link>
          </div>
        </div>
      </form>
    </div>
  );
};
