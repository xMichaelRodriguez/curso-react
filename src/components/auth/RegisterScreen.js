import React from "react";
import { Link } from "react-router-dom";

export const RegisterScreen = () => {
  return (
    <>
      <h3 className="auth__title mb-5">Register</h3>
      <form>
        <div className="auth__group">
          <input
            type="text"
            className="auth__input"
            name="firstName"
            placeholder="Your  First Name"
            autoComplete="false"
          />
          <span class="highlight"></span>
          <span class="bar"></span>
          <label>First Name</label>
        </div>
        <div className="auth__group">
          <input
            type="text"
            className="auth__input"
            name="lastName"
            placeholder="Your Last Name"
            autoComplete="false"
          />
          <span class="highlight"></span>
          <span class="bar"></span>
          <label>Last Name</label>
        </div>
        <div className="auth__group">
          <input
            type="email"
            className="auth__input"
            name="email"
            placeholder="Your Email"
            autoComplete="false"
          />
          <span class="highlight"></span>
          <span class="bar"></span>
          <label>Email</label>
        </div>
        <div className="auth__group">
          <input
            type="Password"
            className="auth__input"
            name="password"
            placeholder="Your Password"
          />
          <span class="highlight"></span>
          <span class="bar"></span>
          <label>Password</label>
        </div>

        <button type="submit" className="btn btn-primary btn-block mb-5">
          Sign In
        </button>

        <Link to="/auth/login" className="link">
          Already registered?
        </Link>
      </form>
    </>
  );
};
