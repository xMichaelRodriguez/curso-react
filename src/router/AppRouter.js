import React from "react";
import {
  BrowserRouter as Router,
  Switch,
  Route,
  Redirect,
} from "react-router-dom";
import { HomeScreen } from "../components/HomeScreen";
import { Nabvar } from "../components/Nabvar";
import { NewFileScreen } from "../components/NewFileScreen";
export const AppRouter = () => {
  return (
    <Router>
      <div>
        <Nabvar />
        <div className="container-fluid mt-3">
          <Switch>
            <Route exact path="/home" component={HomeScreen} />
            <Route exact path="/newFile" component={NewFileScreen} />
            <Redirect to="/home" />
          </Switch>
        </div>
      </div>
    </Router>
  );
};
