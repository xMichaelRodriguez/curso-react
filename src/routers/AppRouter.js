import React from "react";

import {
  BrowserRouter as Router,
  Redirect,
  Route,
  Switch,
} from "react-router-dom";

import { JournalScren } from "../components/journal/JournalScren";
import { AuthRouter } from "./AuthRouter";
export const AppRouter = () => {
  return (
    <Router>
      <div>
        <Switch>
          <Route path="/auth" component={AuthRouter} />
          <Route exact path="/" component={JournalScren} />

          <Redirect to="/auth/login" />
        </Switch>
      </div>
    </Router>
  );
};
