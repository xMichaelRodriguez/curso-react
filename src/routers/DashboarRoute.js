import React from "react";
import { Redirect, Route, Switch } from "react-router-dom";
import { DcScreen } from "../components/dc/DcScreen";
import { HeroesScreen } from "../components/heroes/HeroesScreen";
import { Navbar } from "../components/iu/NavBar";
import { MarvelScreen } from "../components/marvel/MarvelScreen";

export const DashboarRoute = () => {
  return (
    <>
      <Navbar />

      <div className="container mt-2">
        <Switch>
          <Route exact path="/marvel" component={MarvelScreen} />
          <Route exact path="/hero/:heroeId" component={HeroesScreen} />
          <Route exact path="/dc" component={DcScreen} />

          <Redirect to="/marvel" />
        </Switch>
      </div>
    </>
  );
};
