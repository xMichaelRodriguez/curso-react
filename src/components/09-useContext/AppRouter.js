import React from "react";
import { BrowserRouter as Router, Switch, Route, Redirect } from "react-router-dom";
import { NavBar } from "./NavBar";

import { AboutPage } from "./AboutPage";
import { LoginScreen } from "./LoginScreen";
import { HomeScreen } from "./HomeScreen";
export const AppRouter = () => {
  return (
    <Router>
      <div>
        <NavBar />

       <div className="container">
       <Switch>
          <Route exact path="/" component={HomeScreen} />
          <Route exact path="/about" component={AboutPage} />
          <Route exact path="/login" component={LoginScreen} />
          <Redirect to="./" />
        </Switch>
       </div>
      </div>
    </Router>
  );
};
