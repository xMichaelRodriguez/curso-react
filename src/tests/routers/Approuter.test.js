import React from 'react';
import '@testing-library/jest-dom/extend-expect';
import {   render } from '@testing-library/react';
import { AppRouter } from '../../routers/AppRouter';
import { AuthContext } from '../../auth/AuthContext';


describe('Test on component <AppRouter />', () => {
    
    test('It should show the login component if you are not authenticated', () => {
        const contextValue = {
            dispatch: jest.fn(),
            user: {
                logged: false
            }
        }
        const component = render(
            <AuthContext.Provider value={contextValue}>
                <AppRouter />
            </AuthContext.Provider>
        );
        const el = component.container;
           
        expect(el).toMatchSnapshot()
  
    });
    test('It should show the marvel component if you are  authenticated', () => {
        const contextValue = {
            dispatch: jest.fn(),
            user: {
                logged: true,
                name:"Michael Rodriguez"
            }
        }
        const component = render(
            <AuthContext.Provider value={contextValue}>
                <AppRouter />
            </AuthContext.Provider>
        );
        
       component.getByText('Asociaciones')  
       
  
    });
    
});
