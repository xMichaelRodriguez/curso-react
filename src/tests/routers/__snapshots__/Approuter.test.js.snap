// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test on component <AppRouter /> It should show the login component if you are not authenticated 1`] = `
<div>
  <div>
    <div
      class="container mt-5"
    >
      <h1>
        Login
      </h1>
      <hr />
      <button
        class="btn btn-primary"
      >
        Login
      </button>
    </div>
  </div>
</div>
`;
